src/boat_simulator/boat_simulator/nodes/low_level_control/controller.py

```diff
@@ -0,0 +1,259 @@
+        """
+        self.current_control_ang = current_control_ang
+        self.time_step = time_step
+        self.control_speed = control_speed
+        self.max_angle_range = max_angle_range
+        self.running_error = 0.0
+
+    def update_state(self) -> bool:
+        """Updates the controller position iteratively based on control_speed * time_step, until
+        the control target angle is reached.
+
+        Args:
+            `max_angle_range` (tuple): Max angle range individual to each different controller
+
+
+        Returns:
+            False if target control angle was not reached, else
+            True if target control angle has been reached (running error = 0)
+        """
+        error_tracker = False
+        if np.isclose(self.running_error, 0, 0.01):
+            return True
+        else:
+            change = copysign(self.control_speed * self.time_step, self.running_error)
+
+        if abs(self.running_error) > abs(change):
+            next_control = max(self.current_control_ang + change, self.max_angle_range[0])
+            self.running_error -= change
+        else:
+            next_control = max(
+                self.current_control_ang + self.running_error, self.max_angle_range[0]
+            )
+            self.running_error = 0
+            error_tracker = True
+
+        self.current_control_ang = min(next_control, self.max_angle_range[1])
+        return error_tracker
+
+
+class RudderController(ActuatorController):
+    """General Class for the Actuator Controller.
+
+    RudderController Extends: ActuatorController
+    """
+
+    def __init__(
+        self,
+        current_heading: Scalar,
+        desired_heading: Scalar,
+        current_control_ang: Scalar,
+        time_step: Scalar,
+        kp: Scalar,
+        cp: Scalar,
+        control_speed: Scalar,
+        max_angle_range=None,
+    ):
+        """Initializes the class attributes.
+
+        Args:
+            `current_heading` (Scalar): Current boat heading direction in degrees,
+             0 degrees (North) at the positive y axis and increasing clockwise.
+            `desired_heading` (Scalar): Target boating heading direction in degrees,
+             0 degrees (North) at the positive y axis and increasing clockwise.
+            `current_control_ang` (Scalar): Current control mechanism angle in degrees.
+            `time_step` (Scalar): Time per iteration given in seconds.
+            `kp` (Scalar): Proportional constant when calculating error.
+            `cp` (Scalar): Tuning parameter for control action.
+            `control_speed` (Scalar): Speed of controller change in degrees per second.
+            'running_error' (Scalar): Error between current and target control angle.
+
+        """
+
+        super().__init__(
+            current_control_ang,
+            time_step,
+            control_speed,
+            max_angle_range,
+        )
+
+        self.max_angle_range = RUDDER_MAX_ANGLE_RANGE
```

Comment:
If you address the comment above, this line is no longer necessary because the super class constructor already does the assignment for you!